src/bananabot.py

```diff
@@ -182,80 +182,156 @@ def toggle_status():
+templates = Jinja2Templates(directory="templates")
+
+@app.on_event("startup")
+async def startup_event():
+    global password
+
+    # Generate random password and print it, then encode
+    password = ''.join(random.choice(string.ascii_letters) for i in range(10))
+    print(f"Admin Password: { password }")
+
+    # Hash the password and store the hash
+    password = bcrypt.hashpw(password.encode("utf8"), salt)
+
+
+#### API Endpoints ####
+@app.post('/toggle-status', status_code=status.HTTP_200_OK)
+def update_status(dependencies = Depends(get_current_user)):
+    return toggle_status()
+
+
+@app.get('/banana-time', status_code=status.HTTP_200_OK)
+def get_banana_time():
+    return Announcement.banana_time
+
+
+@app.post('/banana-time', status_code=status.HTTP_200_OK)
+def post_banana_time(banana_time_data: BananaTimeData, dependencies = Depends(get_current_user)):
+    set_banana_time(banana_time_data.time)
+    return Announcement.banana_time
+
+
+@app.post('/banana-text', status_code=status.HTTP_200_OK)
+def post_banana_text(banana_time_data: BananaTimeData, dependencies = Depends(get_current_user)):
+    set_banana_time_text(banana_time_data.text)
+    return announcements['banana_time'].text
+
+
+@app.get('/selected-days', status_code=status.HTTP_200_OK)
+def get_selected_days(dependencies = Depends(get_current_user)):
+    return Announcement.selected_days
+
+
+@app.post('/selected-days', status_code=status.HTTP_200_OK)
+def post_selected_days(new_days: SelectedDaysData, dependencies = Depends(get_current_user)):
+    new_days = {
+        "monday": new_days.monday == "on",
+        "tuesday": new_days.tuesday == "on",
+        "wednesday": new_days.wednesday == "on",
+        "thursday": new_days.thursday == "on",
+        "friday": new_days.friday == "on",
+        "saturday": new_days.saturday == "on",
+        "sunday": new_days.sunday == "on"
+    }
+
+    update_selected_days(new_days)
+    return Announcement.selected_days
+
+
+@app.get('/announcements')
+def get_announcements(dependencies = Depends(get_current_user)):
+    return announcements
+
+
+@app.post('/announcements', status_code=status.HTTP_201_CREATED)
+def post_announcements(announcement: AnnouncementData, dependencies = Depends(get_current_user)):
+    match announcement.type:
+        case 'time':
+            return announcements[add_time_announcement(announcement.time, announcement.text)]
+        case 'mins_before':
+            return announcements[add_mins_before_announcement(announcement.mins_before, announcement.text)]
+        case 'instant':
+            Announcement.send_message(announcement.text)
+            return
+        
+    # Raise exception if none of the above conditions were satisfied
+
+
+@app.delete('/announcements/{announcement_id}', status_code=status.HTTP_200_OK)
+def delete_announcement(announcement_id: int, dependencies = Depends(get_current_user)):
+    remove_announcement(announcement_id)
```

Comment:
Perhaps we should be checking the return value of `remove_announcement` here? In it's current state, the function could fail and the server would still respond with a `200 OK`. It should probably raise a HTTPException if `remove_announcement` fails and return an appropriate error code